src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -139,127 +252,124 @@ public String getName() {
+        }
 
+        messages.add(ReportingMessage.fromEvent(this, commerceEvent));
 
-    @Override
-    public List<ReportingMessage> logLtvIncrease(BigDecimal bigDecimal, BigDecimal bigDecimal1, String s, Map<String, String> map) {
-        return null;
+        return messages;
     }
 
-    @Override
-    public List<ReportingMessage> logEvent(CommerceEvent commerceEvent) {
-        List<ReportingMessage> messages = new LinkedList<ReportingMessage>();
-        if (!commerceEvent.getProductAction().equals(Product.PURCHASE)) {
-            List<MPEvent> eventList = CommerceEventUtils.expand(commerceEvent);
-            if (eventList != null) {
-                for (int i = 0; i < eventList.size(); i++) {
-                    try {
-                        logEvent(eventList.get(i));
-                        messages.add(ReportingMessage.fromEvent(this, commerceEvent));
-                    } catch (Exception e) {
-                        Logger.warning("Failed to call logCustomEvent to Singular kit: " + e.toString());
-                    }
-                }
-            }
-        } else {
-            if (!KitUtils.isEmpty(commerceEvent.getCurrency())) {
-                currency = commerceEvent.getCurrency();
-            }
-            if (commerceEvent.getProducts().size() > 0) {
-                List<Product> productList = commerceEvent.getProducts();
-                for (Product product : productList) {
-                    productPrice = product.getUnitPrice();
-                    productQuantity = product.getQuantity();
-                    if (!KitUtils.isEmpty(product.getSku())) {
-                        productSKU = product.getSku();
-                    }
-                    if (!KitUtils.isEmpty(product.getCategory())) {
-                        productCategory = product.getCategory();
+    private List<ReportingMessage> handleNonPurchaseEvents(CommerceEvent commerceEvent) {
+        List<ReportingMessage> messages = new ArrayList<>();
+
+        // Getting the mParticle events from the commerce event
+        List<MPEvent> eventList = CommerceEventUtils.expand(commerceEvent);
+        if (eventList != null) {
+            for (MPEvent event : eventList) {
+                try {
+                    for (ReportingMessage message : logEvent(event)) {
+                        messages.add(message);
                     }
-                    productName = product.getName();
-                    amount = product.getTotalAmount();
-                    Singular.revenue(currency, amount, productSKU, productName, productCategory, (int) productQuantity, productPrice);
+                } catch (Exception e) {
+                    Logger.warning("Failed to call logCustomEvent to Singular kit: " + e.toString());
                 }
-                messages.add(ReportingMessage.fromEvent(this, commerceEvent));
             }
         }
+
+        messages.add(ReportingMessage.fromEvent(this, commerceEvent));
```

Comment:
1 small thing, you would want to add _either_ the `CommerceEvent` or the `List<MPEvent>` generated by the expansion to the `List<ReportingMessage>`. I can't think of any functional difference between the two, since what your doing is basically recreating the code we would call for non `CommerceListener` Kit's for non `Purchase` CommerceEvent's, but you should either remove this line, or the one where you're adding message from the expanded CommerceEvent